app/keepers/keepers.go

```diff
@@ -358,12 +362,19 @@ func NewAppKeeper(
 	// If evidence needs to be handled for the app, set routes in router here and seal
 	appKeepers.EvidenceKeeper = *evidenceKeeper
 
+	appKeepers.IBCFeeKeeper = ibcfeekeeper.NewKeeper(
+		appCodec, appKeepers.keys[ibcfeetypes.StoreKey],
+		appKeepers.IBCKeeper.ChannelKeeper, // may be replaced with IBC middleware
```

Comment:
Shouldn't this be the PFM module? Similarly how PFM has the rate limit module as an ICS4Wrapper

Reply:
Also, the transferKeeper should have the ibcfee as an ICS4Wrapper instead of the PFM module. Right?

Reply:
The wiring from this PR should be correct:
* https://github.com/cosmos/gaia/pull/3038/commits/246ad288375d6b6e4e2fe068a533074dea7e3700

```go
appKeepers.IBCFeeKeeper = ibcfeekeeper.NewKeeper(
	...
	appKeepers.IBCKeeper.ChannelKeeper, // ICS4Wrapper
	...
)

appKeepers.ICAHostKeeper = icahostkeeper.NewKeeper(
	...
	apKeepers.IBCFeeKeeper, // ICS4Wrapper
)

appKeepers.RatelimitKeeper = *ratelimitkeeper.NewKeeper(
	...
	appKeepers.IBCFeeKeeper, // ICS4Wrapper
)

// ICA Controller keeper
appKeepers.ICAControllerKeeper = icacontrollerkeeper.NewKeeper(
	...
	appKeepers.IBCFeeKeeper, // ICS4Wrapper
)

// PFMRouterKeeper must be created before TransferKeeper
appKeepers.PFMRouterKeeper = pfmrouterkeeper.NewKeeper(
	...
	appKeepers.RatelimitKeeper, // ICS4Wrapper
)

appKeepers.TransferKeeper = ibctransferkeeper.NewKeeper(
	...
	appKeepers.PFMRouterKeeper, // ISC4 Wrapper: PFM Router middleware
)

// stacks \\

// * RecvPacket -> IBC core -> Fee -> RateLimit -> PFM -> Transfer (AddRoute)
// * SentPacket -> Transfer -> PFM -> RateLimit -> Fee -> IBC core (ICS4Wrapper)
var transferStack porttypes.IBCModule
transferStack = transfer.NewIBCModule(appKeepers.TransferKeeper)
transferStack = pfmrouter.NewIBCMiddleware(transferStack, appKeepers.PFMRouterKeeper, ...)
transferStack = ratelimit.NewIBCMiddleware(appKeepers.RatelimitKeeper, transferStack)
transferStack = ibcfee.NewIBCMiddleware(transferStack, appKeepers.IBCFeeKeeper)

// ICAHost Stack
var icaHostStack porttypes.IBCModule = icahost.NewIBCModule(appKeepers.ICAHostKeeper)
icaHostStack = ibcfee.NewIBCMiddleware(icaHostStack, appKeepers.IBCFeeKeeper)

// ICAController Stack
var icaControllerStack porttypes.IBCModule = icacontroller.NewIBCMiddleware(nil, appKeepers.ICAControllerKeeper)
icaControllerStack = ibcfee.NewIBCMiddleware(icaControllerStack, appKeepers.IBCFeeKeeper)

```